upstairs/src/client.rs

```diff
@@ -1508,6 +1512,16 @@ impl DownstairsClient {
                     }
                     self.last_flush = ds_id;
                 }
+                IOop::Barrier { .. } => {
+                    assert!(read_data.blocks.is_empty());
+                    assert!(read_data.data.is_empty());
+                    assert!(extent_info.is_none());
+
+                    if jobs_completed_ok == 2 {
```

Comment:
For the barrier, do we want `jobs_completed_ok == 3`?

And, we are not acking this to anyone, but do you plan to use ackable later to indicate that future action can move forward?

Reply:
I'm ambivalent!  Like you said, we're not actually acking this to anyone; I also don't plan to use presence in `ackable_work` as an indication for anything here (unlike how it's used in live-repair).

This means the only thing that happens when the job is acked is that it moves from `GuestWork::active` to `GuestWork::completed`.  Setting the ack to happen at `jobs_completed_ok == 3` seems reasonable to me.

Reply:
If we are not using `ackable` for anything, then I would vote for not setting it at all.
Otherwise I'll come back here and see it set and wonder what was consuming that and be confused when I found nothing.

I do care about the dtrace probe though, that one we might actually want to know when this operation has finised on all three downstairs. I don't think I would ever care to know that the barrier had finished on 2/3 downstairs.

Reply:
We do need to use `ackable`, because the job is stored in `GuestWork` so we need to "ack" it to remove it from that map (even though nothing else happens in the ack).

I changed the threshold to 3 in ed52bf5